Sources/SwiftKuery/Delete.swift

```diff
@@ -48,7 +51,24 @@ public struct Delete: Query {
         if syntaxError != "" {
             throw QueryError.syntaxError(syntaxError)
         }
-        var result = try "DELETE FROM " + table.build(queryBuilder: queryBuilder)
+        
+        var result = ""
+        
+        if let with = with {
+            result += "WITH "
+                + "\(try with.map { try $0.buildWith(queryBuilder: queryBuilder) }.joined(separator: ", "))"
+                + " "
+        }
+        
+        result += "DELETE FROM "
+        
+        result += try table.build(queryBuilder: queryBuilder)
+        
+        if queryBuilder.withDeleteRequiresUsing,
+            let with = with {
```

Comment:
Could you please switch between the checks: it makes more sense if we check how to build WITH only after verifying that it exists.